docs/src/pages/playground.tsx

```diff
@@ -0,0 +1,212 @@
+import React, { useEffect, useRef, useState } from 'react'
+import Layout from '@theme/Layout'
+import sdk from '@stackblitz/sdk'
+import styles from './playground.module.css'
+import { useColorMode } from '@docusaurus/theme-common'
+
+interface Project {
+	categoryTitle: string
+	subCategory: Array<CategoryItems>
+}
+
+interface CategoryItems {
+	name: string
+	templates: Array<Templates>
+}
+
+interface Templates {
+	title: string
+	desc: string
+	link: string
+	idx: number
+}
+
+function Playground() {
+	const [projectLink, setProjectLink] = useState('Wave-Play/robo.js/tree/main/templates/starter-app-js-react')
+	const [dropdown, setDropdown] = useState(false)
+	const [embedLoading, setEmbedLoading] = useState(true)
+	const { colorMode } = useColorMode()
+	const [fetchError, setFetchError] = useState<null | string>(null)
+	const [templates, setTemplates] = useState<Project[] | null>(null)
+	const [selectedTemplateIndex, setSelectedTemplateIndex] = useState(0)
+
+	const embedDiv = useRef<HTMLIFrameElement | null>(null)
+	const editor = useRef(null)
+
+	useEffect(() => {
+		const fetchTemplates = async () => {
+			const res = await fetch('http://localhost:3000/api/templates')
+
+			if (res.status !== 20) {
+				setFetchError(res.statusText)
+			}
+
+			if (res.ok && res.status === 200) {
+				const json = await res.json()
+				setTemplates(json)
+			}
+		}
+		fetchTemplates()
+	}, [])
+
+	useEffect(() => {
+		if (editor && editor.current) {
+			editor.current.setTheme(colorMode)
+		}
+	}, [colorMode])
+
+	const clickHandler = () => {
+		if (dropdown) {
+			setDropdown(false)
+		}
+	}
+	useEffect(() => {
+		document.addEventListener('click', clickHandler)
+
+		return () => {
+			document.removeEventListener('click', clickHandler)
+		}
+	}, [])
+
+	useEffect(() => {
+		if (embedDiv && embedDiv.current) {
+			sdk
+				.embedGithubProject('embed', projectLink, {
+					devToolsHeight: 40,
+					showSidebar: true,
+					terminalHeight: 40,
+					theme: colorMode,
+					view: 'default'
+				})
+				.then((v) => {
+					setEmbedLoading(false)
+					editor.current = v.editor
+				})
+		}
+	}, [projectLink])
+
+	function renderSearchBar() {
```

Comment:
Wouldn't this be cleaner and more concise inline?

Reply:
I am not sure what you mean by that

Reply:
Moving the contents of `renderSearchBar` into where the function is called.

Considering it's only called in one place, I think it makes more sense to put the contents inline in the parents return JSX so it maintains context. The way it is now almost looks like the search bar is a different reusable component.